core/tests/gateway_e2e/dependencies.cmake

```diff
@@ -36,4 +36,9 @@ if(NOT EXISTS ${PROJECT_SOURCE_DIR}/deps/iot-sdk-c/parson/README.md)
         message(FATAL_ERROR "Error pulling parson submodule: ${res}")
     endif()
 endif()
-findAndInstall(azure_iot_sdks 1.1.5 ${PROJECT_SOURCE_DIR}/deps/iot-sdk-c ${PROJECT_SOURCE_DIR}/deps/iot-sdk-c -Duse_installed_dependencies=ON -Dbuild_as_dynamic=ON -Drun_e2e_tests=ON -Duse_openssl=OFF -G "${CMAKE_GENERATOR}")
+if(${run_e2e_tests})
+set(iothub_run_e2e ON)
+else()
+set(iothub_run_e2e OFF)
+endif()
+findAndInstall(azure_iot_sdks 1.1.5 ${PROJECT_SOURCE_DIR}/deps/iot-sdk-c ${PROJECT_SOURCE_DIR}/deps/iot-sdk-c -Duse_installed_dependencies=ON -Dbuild_as_dynamic=ON -Drun_e2e_tests=${iothub_run_e2e} -Duse_openssl=OFF -G "${CMAKE_GENERATOR}")
```

Comment:
I was thinking this might be more complicated, but I think just passing ${run_e2e_tests} will work here.